modules/set-the-default-max-container-root-partition-size-for-overlay-with-crio.adoc

```diff
@@ -0,0 +1,122 @@
+// Module included in the following assemblies:
+//
+// post_installation_configuration/machine-configuration-tasks.adoc
+
+[id="set-the-default-max-container-root-partition-size-for-overlay-with-crio_{context}"]
+= Setting the default maximum container root partition size for Overlay with CRI-O
+
+The root partition of each container shows all of the available disk space of the underlying host. Follow this guidance to set a maximum partition size for the root disk of all containers.
+
+To configure the maximum Overlay size, as well as other CRI-O options like the log level and PID limit, you can create the following `ContainerRuntimeConfig` CRD:
+
+[source,yaml]
+----
+apiVersion: machineconfiguration.openshift.io/v1
+kind: ContainerRuntimeConfig
+metadata:
+ name: overlay-size
+spec:
+ machineConfigPoolSelector:
+   matchLabels:
+     custom-crio: overlay-size
+ containerRuntimeConfig:
+   pidsLimit: 2048
+   logLevel: debug
+   overlaySize: 8G
+----
+
+.Procedure
+
+. Create the configuration object:
++
+[source,terminal]
+----
+$ oc apply -f overlaysize.yml
+----
+
+. To apply the new CRI-O configuration to your worker nodes, edit the worker machine config pool:
++
+[source,terminal]
+----
+$ oc edit machineconfigpool worker
+----
+
+. Add the `custom-crio` label based on the `matchLabels` name you set in the `ContainerRuntimeConfig` CRD:
++
+[source,yaml]
+----
+apiVersion: machineconfiguration.openshift.io/v1
+kind: MachineConfigPool
+metadata:
+  creationTimestamp: "2020-07-09T15:46:34Z"
+  generation: 3
+  labels:
+    custom-crio: overlay-size
+    machineconfiguration.openshift.io/mco-built-in: ""
+----
+
+. Once saved, view the machine configs:
++
+[source,terminal]
+----
+$ oc get machineconfigs
+----
++
+You will see a new `99-worker-zxy-containerruntime` object as well as a new `rendered-worker-xyz` object created:
++
+.Example output
+[source,terminal]
+----
+99-worker-xyz-containerruntime   4173030d89fbf4a7a0976d1665491a4d9a6e54f1   2.2.0             7m42s
```

Comment:
The MC names changed some time back, this should be `99-worker-generated-containerruntime` now.

Reply:
Updated. Thank you!